File: java/cdk8s-crd/src/main/java/imports/io/cert_manager/CertificateSpecSignatureAlgorithm.java
package imports.io.cert_manager;

/**
 * Signature algorith to use.
 * <p>
 * Allowed values for RSA keys: SHA256WithRSA, SHA384WithRSA, SHA512WithRSA.
 * Allowed values for ECDSA keys: ECDSAWithSHA256, ECDSAWithSHA384, ECDSAWithSHA512.
 * Allowed values for Ed25519 keys: PureEd25519.
 */
@javax.annotation.Generated(value = "jsii-pacmak/1.86.1 (build defb235)", date = "2025-04-23T10:37:21.380Z")
@software.amazon.jsii.Jsii(module = imports.io.cert_manager.$Module.class, fqn = "iocert-manager.CertificateSpecSignatureAlgorithm")
public enum CertificateSpecSignatureAlgorithm {
    /**
     * SHA256WithRSA.
     */
    SHA256_WITH_RSA,
    /**
     * SHA384WithRSA.
     */
    SHA384_WITH_RSA,
    /**
     * SHA512WithRSA.
     */
    SHA512_WITH_RSA,
    /**
     * ECDSAWithSHA256.
     */
    ECDSA_WITH_SHA256,
    /**
     * ECDSAWithSHA384.
     */
    ECDSA_WITH_SHA384,
    /**
     * ECDSAWithSHA512.
     */
    ECDSA_WITH_SHA512,
    /**
     * PureEd25519.
     */
    PURE_ED25519,
}

File: go/cdk8s-crd/imports/certmanagerio/CertificateSpecSignatureAlgorithm.go
package certmanagerio


// Signature algorith to use.
//
// Allowed values for RSA keys: SHA256WithRSA, SHA384WithRSA, SHA512WithRSA.
// Allowed values for ECDSA keys: ECDSAWithSHA256, ECDSAWithSHA384, ECDSAWithSHA512.
// Allowed values for Ed25519 keys: PureEd25519.
type CertificateSpecSignatureAlgorithm string

const (
	// SHA256WithRSA.
	CertificateSpecSignatureAlgorithm_SHA256_WITH_RSA CertificateSpecSignatureAlgorithm = "SHA256_WITH_RSA"
	// SHA384WithRSA.
	CertificateSpecSignatureAlgorithm_SHA384_WITH_RSA CertificateSpecSignatureAlgorithm = "SHA384_WITH_RSA"
	// SHA512WithRSA.
	CertificateSpecSignatureAlgorithm_SHA512_WITH_RSA CertificateSpecSignatureAlgorithm = "SHA512_WITH_RSA"
	// ECDSAWithSHA256.
	CertificateSpecSignatureAlgorithm_ECDSA_WITH_SHA256 CertificateSpecSignatureAlgorithm = "ECDSA_WITH_SHA256"
	// ECDSAWithSHA384.
	CertificateSpecSignatureAlgorithm_ECDSA_WITH_SHA384 CertificateSpecSignatureAlgorithm = "ECDSA_WITH_SHA384"
	// ECDSAWithSHA512.
	CertificateSpecSignatureAlgorithm_ECDSA_WITH_SHA512 CertificateSpecSignatureAlgorithm = "ECDSA_WITH_SHA512"
	// PureEd25519.
	CertificateSpecSignatureAlgorithm_PURE_ED25519 CertificateSpecSignatureAlgorithm = "PURE_ED25519"
)

